app/controllers/products_controller.rb

```diff
@@ -0,0 +1,124 @@
+
+
+class ProductsController < ApplicationController
+  include ApplicationHelper
+  before_action :navbar_categories, only: [:index]
+  before_action :current_user_owns_product, only: [:update, :delete, :retire]
+  before_action :find_product, only: [:retire, :buy, :show, :destroy]
+
+
+
+  def buy
+    #if there is not yet an order_id in session, make it now
+    #this is totally independent of being logged in.
+    #then if the id is nil, make an order and put it's id in the session hash
+    #if there already is an order in the session, add the product to it.
+      @current_order = current_order
```

Comment:
This assignment isn't necessary because your current_order method defines @current_order.